src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1272,6 +1272,73 @@ object ICustomAstVisitor.VisitFileRedirection(FileRedirectionAst fileRedirection
 
         private void InferTypesFrom(CommandAst commandAst, List<PSTypeName> inferredTypes)
         {
+            if (commandAst.Redirections.Count > 0)
+            {
+                var mergedStreams = new HashSet<RedirectionStream>();
+                bool allStreamsMerged = false;
+                foreach (RedirectionAst streamRedirection in commandAst.Redirections)
+                {
+                    if (streamRedirection is FileRedirectionAst fileRedirection)
+                    {
+                        if (fileRedirection.FromStream is RedirectionStream.All or RedirectionStream.Output)
+                        {
+                            // command output is redirected so it returns nothing.
+                            return;
```

Comment:
To continue previous question. Is it possible to have file and non-file redirections in the same time? If no, we can remove the branch and check only MergingRedirectionAst. If yes, we should change all the code.

Reply:
As mentioned in the other response, yes you can have both: `ls C:\, L:\ *>&1 1>$HOME\MyData.txt`.

Reply:
It was my misunderstanding of redirection combinations. Thanks for clarify!